File: schemaorggo/rating.go
package schemaorggo

import "encoding/json"

// Rating see : https://schema.org/Rating
type Rating struct {
	Intangible

	typeContext
}

func (v Rating) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.Intangible.intoMap(intop)

	into := *intop

	*intop = into

	return nil
}

func (v Rating) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "Rating"

	return data, nil
}

func (v Rating) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/rating.php
<?php

class Rating extends Intangible implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'Rating';
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'Rating'
		);
		
		return $out;
	}
}
